api/errors/errors_test.go

```diff
@@ -0,0 +1,64 @@
+package errors
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+)
+
+func newError(errorname string) error {
+
+	return fmt.Errorf("test %v", errorname)
+}
+
+func TestErrors(t *testing.T) {
+	errmsg := newError("testapiError")
+	err := apiError{
+		error:      errmsg,
+		statusCode: 0,
+	}
+	assert.Equal(t, err.HTTPErrorStatusCode(), err.statusCode)
+
+	errmsg = newError("testErrorWithStatusCode")
+	errcode := 1
+	serr := NewErrorWithStatusCode(errmsg, errcode)
+	apierr, a := serr.(apiError)
+	if !a {
+		t.Fatal("excepted err is apiError type")
+	}
+	assert.Equal(t, errcode, apierr.statusCode)
+
+	errmsg = newError("testNewBadRequestError")
+	baderr := NewBadRequestError(err.error)
+	apierr, a = baderr.(apiError)
+	if !a {
+		t.Fatal("excepted err is apiError type")
+	}
+	assert.Equal(t, http.StatusBadRequest, apierr.statusCode)
+
+	errmsg = newError("testRequestForbiddenError")
+	ferr := NewRequestForbiddenError(err.error)
+	apierr, a = ferr.(apiError)
+	if !a {
+		t.Fatal("excepted err is apiError type")
+	}
+	assert.Equal(t, http.StatusForbidden, apierr.statusCode)
+
+	errmsg = newError("testRequestNotFoundError")
+	nerr := NewRequestNotFoundError(err.error)
+	apierr, a = nerr.(apiError)
+	if !a {
+		t.Fatal("excepted err is apiError type")
+	}
+	assert.Equal(t, http.StatusNotFound, apierr.statusCode)
+
+	errmsg = newError("testRequestConflictError")
```

Comment:
@chchliang @AkihiroSuda meant that this `errmsg` variable is not used below, so either something is missing or it should be there 👼